instructor/common.py

```diff
@@ -97,66 +97,71 @@ def assignment_setup(request):
 def course_setup(request):
-    CourseForm = modelform_factory(models.Course, fields=['code', 'name'])
 
-    if request.method == "POST":
-        if 'course_pk' in request.POST:
-            # Adding to an existing course
-            course = models.Course.objects.get(pk=request.POST['course_pk'])
-        else:
-            # Create a new course
+    CourseFormSet = modelformset_factory(models.Course, extra=1, can_delete=True, fields=['name', 'code'])
+    course_selected = 0
+    if request.method == 'POST':
+
+        # change this course's name or code
+        formset = CourseFormSet(request.POST)
+        if formset.is_valid():
+            courses = formset.save(commit=False)
+            for obj in formset.deleted_objects:
+                obj.delete()
             user = User.objects.get(username=request.user)
-            form = CourseForm(request.POST)
-            if form.is_valid():
-                course = form.save(commit=False)
-                course.owner_id = user.id
+            for course in courses:
+                # need to set the owner
+                course.owner = user
                 course.save()
-            else:
-                all_courses = models.Course.objects.all()
-                return render_to_response('instructor/course_setup.html',
-                                          {'courses': all_courses,
-                                           'form': form,
-                                           'new': request.session['new']},
-                                          context_instance=RequestContext(request))
-
-        # Create a new assignment
-        assignment_name = request.session['assignment_name']
-        assignment_id = assignment_name + datetime.datetime.now().strftime("%I:%M%p on %B %d, %Y")
-
-        if request.session['based_on']:
-            based_on = models.Assignment.objects.get(pk=request.session['based_on'])
-            a = models.Assignment(course=course,
-                                  name=assignment_name,
-                                  assignment_id=assignment_id,
-                                  basedOn=based_on)
-        else:
-            a = models.Assignment(course=course,
-                                  name=assignment_name,
-                                  assignment_id=assignment_id)
-        a.save()
-        request.session['assignment_id'] = a.id
-        request.session['new'] = False
-
-        if 'save' in request.POST:
-            return redirect("common_course_modify")
-        else:
-            # Continue
-            return redirect('common_assignments_edit_strains')
 
+    # if there is at least one course
+    all_courses = models.Course.objects.all()
+    if not course_selected and len(all_courses) > 0:
+        course_selected = all_courses[0]
+        create_assignment(request, course_selected)
+        if 'continue' in request.POST:
+            return redirect("common_assignments_edit_strains")
+
+    if course_selected:
```

Comment:
At this point `course_selected` looks like it's always set to something so you may want to get rid of the check and the rest of the function after the redirect.